ComponentKit/TransactionalDataSources/Common/CKTransactionalComponentDataSource.mm

```diff
@@ -157,11 +157,23 @@ - (void)componentScopeHandleWithIdentifier:(CKComponentScopeHandleIdentifier)glo
       [self _processStateUpdates];
     });
   }
+  if (mode == CKUpdateModeAsynchronous) {
+    _pendingAsynchronousStateUpdates[rootIdentifier].insert({globalIdentifier, updateBlock});
+  } else {
+    _pendingSynchronousStateUpdates[rootIdentifier].insert({globalIdentifier, updateBlock});
+  }
+}
 
+- (void)componentScopeHandleWithIdentifier:(CKComponentScopeHandleIdentifier)globalIdentifier
+                            rootIdentifier:(CKComponentScopeRootIdentifier)rootIdentifier
+         didReceiveStateUpdateToBeEnqueued:(id (^)(id))updateBlock
+                                      mode:(CKUpdateMode)mode
+{
+  CKAssertMainThread();
   if (mode == CKUpdateModeAsynchronous) {
-    _pendingAsynchronousStateUpdates[rootIdentifier].insert({globalIdentifier, stateUpdate});
+    _pendingAsynchronousStateUpdates[rootIdentifier].insert({globalIdentifier, updateBlock});
   } else {
-    _pendingSynchronousStateUpdates[rootIdentifier].insert({globalIdentifier, stateUpdate});
+    _pendingSynchronousStateUpdates[rootIdentifier].insert({globalIdentifier, updateBlock});
   }
 }
```

Comment:
This definitely doesn't work, you can verify that with my test in the sample app. The problem is this breaks the logic on line 155. Once you enqueue a state update the entire mechanism breaks. To make this work any enqueued updates would need to be tracked separately.